documentation/DE_Specific_Service_Proxies.md

```diff
@@ -0,0 +1,87 @@
+# Service proxies
+
+## Use case
+Services defined within a Dynamic Extension (Bundle), run within their own private Spring ApplicationContext.
+
+This means they are not visible from the global Alfresco Spring context. (much like Alfresco subsystems)
+
+In some cases you want your service to be visible to other beans running in the global Alfresco context:
+* you want to override an existing bean
+* you want to integrate with a non-dynamic extension
+
+## Proxy to the rescue
+Alfresco subsystems like the Solr context, are wired into the global context via a Proxy: a proxy bean is defined in the global context and configured with the target bean name and target subcontext. (search -> solr/search)
+
+This approach can almost be reused for Dynamic Extensions. We configure a proxy with a service filter (ie. all beans implementing com.IReplicator) and all calls to that proxy are forwarded to the embedded service.
+
+Note: some exposure to OSGi terminology lies ahead
+
+## Caveats
+Unlike Alfresco subsystems, Dynamic Extensions are started asynchronously + they come and go as they please: you can uninstall, update or install alternative implementations of a particular service. So we cannot guarantee that a service will be available when the proxy is invoked.
+
+Dependencies can be expressed between extensions, but not between the 2 worlds. (Alfresco global context & DE) 
+
+This is a limitation we have to accept. Any calls to the proxy, when the service is not available, will fail with an `IllegalStateException`
+
+## Registering a proxy
+### Publication
+By default, any bean tagged with `@Component` is registered within the local Spring context, but remains private to the extension.
+
+An additional step is required to export it: `@OsgiService`
+```java
+@Component
+@OsgiService
+public class DefaultReplicator implements IReplicator {
+   ...
+}
+```
+
+### Consumption
+In an Alfresco x-context.xml file, register a proxy:
```

Comment:
Might be worth it to clarify that the file doesn't need to be literally named "x-context.xml", e.g.:
>In an Alfresco X-context.xml file (where X is the name of your service), register a proxy: